arbnode/batch_poster.go

```diff
@@ -591,12 +613,85 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 	forcePostBatch := time.Since(nextMessageTime) >= config.MaxBatchPostDelay
 	haveUsefulMessage := false
 
+	var l1BoundMaxBlockNumber uint64 = math.MaxUint64
+	var l1BoundMaxTimestamp uint64 = math.MaxUint64
+	var l1BoundMinBlockNumber uint64
+	var l1BoundMinTimestamp uint64
+	hasL1Bound := config.L1BlockBound != l1BlockBoundIgnore
+	if hasL1Bound {
+		var l1Bound *types.Header
+		var err error
+		if config.L1BlockBound == l1BlockBoundLatest {
+			l1Bound, err = b.l1Reader.LastHeader(ctx)
+		} else if config.L1BlockBound == l1BlockBoundSafe {
+			l1Bound, err = b.l1Reader.LatestSafeBlockHeader(ctx)
+		} else {
+			if config.L1BlockBound != l1BlockBoundFinalized {
```

Comment:
I think we should keep this if statement just for extra safety against a bug, but I'll switch to parsing into an enum too

Reply:
e.g. if we somehow didn't call validate and didn't assign to the l1BlockBound field, it's nice to get an error log line instead of the batch poster just continuing on with weird functionality